addon/index.js

```diff
@@ -0,0 +1,168 @@
+import { guidFor } from '@ember/object/internals';
+import { assert } from '@ember/debug';
+
+/**
+  This class is used internally by Ember and Ember Data.
+  Please do not use it at this time. We plan to clean it up
+  and add many tests soon.
+
+  @class OrderedSet
+  @constructor
+  @private
+*/
+export default class OrderedSet {
+  constructor() {
+    this.clear();
+  }
+
+  /**
+    @method create
+    @static
+    @return {OrderedSet}
+    @private
+  */
+  static create() {
+    let Constructor = this;
+    return new Constructor();
+  }
+
+  /**
+    @method clear
+    @private
+  */
+  clear() {
+    this.presenceSet = Object.create(null);
+    this.list = [];
+    this.size = 0;
+  }
+
+  /**
+    @method add
+    @param obj
+    @param guid (optional, and for internal use)
+    @return {OrderedSet}
+    @private
+  */
+  add(obj, _guid) {
+    let guid = _guid || guidFor(obj);
+    let presenceSet = this.presenceSet;
+    let list = this.list;
+
+    if (presenceSet[guid] !== true) {
+      presenceSet[guid] = true;
+      this.size = list.push(obj);
+    }
+
+    return this;
+  }
+
+  /**
+    @since 1.8.0
```

Comment:
Seems like the `@since` docs should be removed

Reply:
Or corrected to 1.0.0 ;P